plugins/module_utils/foreman_helper.py

```diff
@@ -1510,6 +1503,27 @@ def _recursive_dict_keys(a_dict):
     return keys
 
 
+def _recursive_dict_without_none(a_dict, exclude=None):
+    """
+    Remove all entries with `None` value from a dict, recursively.
+    Also drops all entries with keys in `exclude`
```

Comment:
```suggestion
    Also drops all entries with keys in `exclude` in the top level.
```
I agree that we should not drop the aliases from the nested structures.